apps/api/src/domain/models/photograph/commands/create-photograph/photograph-created.event-handler.integration.spec.ts

```diff
@@ -0,0 +1,148 @@
+            // providers: [CommandInfoService, PhotographCreatedEventHandler],
+            imports: [PersistenceModule.forRootAsync(), CommandModule, PhotographCommandsModule],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        await app.init();
+
+        const connectionProvider = app.get(ArangoConnectionProvider);
+
+        databaseProvider = new ArangoDatabaseProvider(connectionProvider);
+
+        // arangoDatabaseForCollection =
+        //     databaseProvider.getDatabaseForCollection('photograph__VIEWS');
+
+        testQueryRepository = new ArangoPhotographQueryRepository(
+            connectionProvider,
+            new ConsoleCoscradCliLogger()
+        );
+    });
+
+    beforeEach(async () => {
+        await databaseProvider.getDatabaseForCollection('photograph__VIEWS').clear();
+    });
+
+    afterAll(async () => {
+        databaseProvider.close();
+    });
+
+    describe(`when handling a photograph created event`, () => {
+        it(`should create the expected photograph`, async () => {
+            // const handler = app.get(PhotographCreatedEventHandler);
+
+            // @ts-expect-error Fix this issue
+            // await handler.handle(photographCreated);
+
+            await new PhotographCreatedEventHandler(testQueryRepository).handle(photographCreated);
+
+            const searchResult = await testQueryRepository.fetchById(photographId);
+
+            expect(searchResult).not.toBe(NotFound);
+
+            const foundPhotograph = searchResult as PhotographViewModel;
+
+            const { name: nameDto, contributions, actions } = foundPhotograph;
+
+            const name = new MultilingualText(nameDto);
+
+            const originalPhotographTitleItem = name.getOriginalTextItem();
+
+            expect(originalPhotographTitleItem.text).toBe(photographTitle);
+
+            expect(originalPhotographTitleItem.languageCode).toBe(languageCode);
+
+            expect(actions).toContain('TAG_RESOURCE');
+            expect(actions).toContain('CREATE_NOTE_ABOUT_RESOURCE');
+            expect(actions).toContain('CONNECT_RESOURCES_WITH_NOTE');
+            expect(actions).toContain('PUBLISH_RESOURCE');
+
+            // expect tags to be empty
+            // expect categories to be empty
+            // expect notes to be empty
+
+            expect(
+                contributions.some(
+                    // this should actually be the name and ID
+                    (c) => c.id === dummyContributor.id
+                )
+            );
+
+            // TODO check the contributor's full name as well
```

Comment:
This is a good oportunity for a test helper. `assertResourceViewHasContributions`